dali/base/dadfs.cpp

```diff
@@ -2738,6 +2738,15 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
         return *t;
     }
 
+    IPropertyTree &queryHistory() const
+    {
+        IPropertyTree *history = root->queryPropTree("History");
+        //It can be null if the file created with an older version
+        if (!history)
+            history = root->setPropTree("History",createPTree("History")); // takes ownership
```

Comment:
TBH - I can't remember the semantics.. I thought if it was calling a non-const method of a member object within a const method (as this is) then it was an error.